nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -257,7 +257,10 @@ def forward_for_export(self, audio_signal, length):
         if isinstance(self.pre_encode, nn.Linear):
             audio_signal = self.pre_encode(audio_signal)
         else:
-            audio_signal, length = self.pre_encode(audio_signal, length)
+            # added in order to prevent slowdown in bfloat16 with CUDNN v8 API
+            # to be removed once the above is fixed in cudnn
+            with torch.cuda.amp.autocast(dtype=torch.float32)
+                audio_signal, length = self.pre_encode(audio_signal, length)
```

Comment:
You need to explicitly cast the input to float here otherwise it does nothing.

Reply:
It is my understanding that there's no need .. e.g. https://pytorch.org/docs/stable/amp.html

Reply:
There is. Autocast only is disabled. If the input to the function is still fp16, then the operation will still be done in fp16. That's why we needed the explicit cast in SE too.

Reply:
See the response and discussion regarding this SE PR - https://github.com/NVIDIA/NeMo/pull/2030

Reply:
Hmm can you print out the input tensor, weight tensor and output tensor dtypes with and without explicit cast and see which one works properly ?